### PR TITLE
feat(radar_threshold_filter): add radar_threshold_filter package

### PR DESCRIPTION
**Pull request by scepter914**
## Description

Add `radar_threshold_filter` package.
This package follow [proposed radar sensing architecture](https://github.com/orgs/autowarefoundation/discussions/2531).

## Tests performed

Tests by rosbag.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  RadarScan output;
+  output.header = radar_data_->header;
+  for (const auto & radar_return : radar_data_->returns) {
+    if (isWithinThreshold(radar_return)) {
+      output.returns.push_back(radar_return);
+    }
+  }
+
+  pub_radar_->publish(output);
+}
```
Comment (yukkysaito) [inline sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp:152]: Please let me know if there is a reason why you are publishing onTimer.
Basically, I think it is better to publish with onData.

Comment (scepter914) [reply sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp]: Fixed at [2824cb0](https://github.com/autowarefoundation/autoware.universe/pull/2300/commits/2824cb07ab6915b1d2f311b017f5cf5dbc83dd64)

File: sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  }
+  return true;
+}
+
+void RadarThresholdFilterNode::onData(const RadarScan::ConstSharedPtr msg)
+{
+  radar_data_ = msg;
+  if (!isDataReady()) {
+    return;
+  }
```
Comment (yukkysaito) [inline sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp:127]: no need?

Comment (scepter914) [reply sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp]: Fixed at [2151bd7](https://github.com/autowarefoundation/autoware.universe/pull/2300/commits/2151bd7cb8097aaeb6490cf4f2bac1dbc61bff15)

Comment (yukkysaito) [reply sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp]: Is the isDataReady function needed?

Comment (scepter914) [reply sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp]: Fixed at [d2f8a8b](https://github.com/autowarefoundation/autoware.universe/pull/2300/commits/d2f8a8bcaec48f559f46f0d32e8c87b111fb7913)

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**